static/js/containers/pages/DashboardPage.js

```diff
@@ -223,7 +225,12 @@ export class DashboardPage extends React.Component<Props, State> {
               <div className="date-summary-text col-12">{dateSummary.text}</div>
             </div>
             <div className="row mt-2">
-              <div className="archived-course-link col-lg-7 col-md-8">
+              <div className="d-flex col-lg-3 col-md-4">
+                {!isProgramCourse && courseRunEnrollment.receipt ? (
+                  <Link to={courseRunEnrollment.receipt}>View Receipt</Link>
```

Comment:
Rather than the whole path being returned from the serializer, you should have only the `Order.id` returned and map the link to `routes.receipt` with match/prop set appropriately.